pdl-compiler/src/backends/rust/parser.rs

```diff
@@ -420,9 +414,11 @@ impl<'a> FieldParser<'a> {
                     // TODO(mgeisler): use
                     // https://doc.rust-lang.org/std/array/fn.try_from_fn.html
                     // when stabilized.
-                    let #id = (0..#count)
-                        .map(|_| #parse_element)
-                        .collect::<Result<Vec<_>, DecodeError>>()?
+                    let mut #id = Vec::with_capacity(#count);
+                    for _n in 0..#count {
```

Comment:
I believe it's more direct to use `_` here: that way you don't get any variable allocated. Right now, I think you're relying on `_n` being optimized away.

```suggestion
                    for _ in 0..#count {
```